### PR TITLE
Update sample server to accept json requests as well

### PR DESCRIPTION
**Pull request by R2wenD2**
This completes server side grpc work. The various clients will need updates to actually work properly with the new server. 

I've also added insecure certs to communicate between the servers. Those should be updated to not be hard coded.

### CODE REVIEW

**Comment by furuholm** [comment]
Tried `curl 'http://localhost:10000/v1alpha1/notes'` but got `http: TLS handshake error from [::1]:51176: tls: first record does not look like a TLS handshake` in the server console. 

`curl 'https://localhost:10000/v1alpha1/notes'` causes `http: TLS handshake error from [::1]:51198: remote error: tls: unknown certificate authority`.

Are the insecure certs just a placeholder or do they serve some other purpose? Maybe this should be solved once we have a working config system to allow users to provide their own PKI configuration? (I have added configurability as part of the [branch](https://github.com/combient/Grafeas/tree/postgres-storage) that addresses #84. This could be backported if you don't want to wait for this PR to be merged.)

File: samples/server/go-server/api/server/cert/certs.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
+// cert is an provides insecure keys.
```
Comment (furuholm) [inline samples/server/go-server/api/server/cert/certs.go]: cert provides insecure keys?

**Comment by R2wenD2** [comment]
@furuholm - you need to pass certs with curl or run curl -k in order for this to work. 

The certs are just a placeholder now. As a follow on to this pr, I'd prefer that we have flags to point to the cert files (which can then be part of our config).

File: samples/server/go-server/api/server/cert/certs.go (hunk 1 above)
Comment (R2wenD2) [reply samples/server/go-server/api/server/cert/certs.go]: Fixed.

File: samples/server/go-server/api/server/main/main.go [hunk 2]
```diff
@@ -15,34 +15,87 @@
...
 	flag.Parse()
+	grafeasEndpoint := fmt.Sprintf("localhost:%d", *port)
+
+	opts := []grpc.ServerOption{
+		grpc.Creds(credentials.NewClientTLSFromCert(cert.Pool(), grafeasEndpoint))}
+	grpcServer := grpc.NewServer(opts...)
+	g := v1alpha1.Grafeas{S: storage.NewMemStore()}
+	pb.RegisterGrafeasServer(grpcServer, &g)
+	opspb.RegisterOperationsServer(grpcServer, &g)
 	log.Printf("Server started on port %v", *port)
```
Comment (skelterjohn) [inline samples/server/go-server/api/server/main/main.go]: this log message will only be true after line 78

**Pull request merged**